core/domain/summary_services.py

```diff
@@ -256,3 +256,18 @@ def get_top_rated_exploration_summary_dicts(language_codes):
         reverse=True)[:NUMBER_OF_TOP_RATED_EXPLORATIONS]
 
     return _get_displayable_exp_summary_dicts(sorted_exp_summaries)
+
+
+def get_recently_published_exploration_summary_dicts(language_codes):
+    """Returns a list of recently published explorations with the given language code.
+    """
+    collect_all_exp_summaries = [
+        exp_summary for exp_summary in
+        exp_services.get_all_exploration_summaries().values()
+        if exp_summary.language_code in language_codes]
+
+    summary_dicts = sorted(collect_all_exp_summaries,
+     key=lambda exp_summary: utils.get_time_in_millisecs(
+        exp_summary.exploration_model_created_on), reverse=True)[:8]
```

Comment:
created_on is not the same as first_published. created_on is when the model instance was first created (which is when the exploration is still private).

However, it looks like ExplorationRightsModel has a first_published_msec property. You could add a query to that model class in order to get the correct exploration ids (as described above), and only then retrieve the corresponding summaries.